flow/model/qrecord_stream.go

```diff
@@ -77,3 +70,14 @@ func (s *QRecordStream) IsSchemaSet() bool {
 func (s *QRecordStream) SchemaChan() <-chan struct{} {
 	return s.schemaLatch
 }
+
+func (s *QRecordStream) Err() error {
+	return s.err
+}
+
+func (s *QRecordStream) Close(err error) {
+	if s.err == nil {
```

Comment:
shouldn't this be != nil and an unconditional close?

Reply:
No, notice that we're checking `s.err` not `err`. `close` panics when called twice on a channel. So this method is designed such that:
1. Close(err) can be called multiple times, QRecordStream handling only tracking first error & only closing channel once
2. Close(nil) can be called unconditionally once (ie, defer) & it won't panic if an error already closed channel

Reply:
thanks, please add comment